code/lib/cli/src/automigrate/fixes/new-frameworks/new-frameworks.ts

```diff
@@ -0,0 +1,528 @@
+        mainConfigPath
+      )}, as the new framework also supports features provided by them:
+      ${addonsToRemove.map((dep) => `- * ${chalk.cyan(dep)}`).join('\n')}
+      `;
+    }
+
+    if (Object.keys(addonOptions).length > 0) {
+      migrationSteps += `- Move the addon options "${chalk.yellow(
+        Object.keys(addonOptions).join(', ')
+      )}" in ${chalk.blue(mainConfigPath)} to the ${chalk.yellow('framework.options')} field.\n`;
+    }
+
+    if (builderConfig) {
+      if (typeof builderConfig === 'string') {
+        migrationSteps += `- Remove the ${chalk.yellow('core.builder')} field in ${chalk.blue(
+          mainConfigPath
+        )}.\n`;
+      } else if (Object.keys(builderConfig.options).length > 0) {
+        migrationSteps += `- Move the ${chalk.yellow('core.builder.options')} field in ${chalk.blue(
+          mainConfigPath
+        )} to ${chalk.yellow('framework.options.builder')} and remove the ${chalk.yellow(
+          'core.builder'
+        )} field.\n`;
+      }
+    }
+
+    if (
+      dependenciesToRemove.includes('@storybook/builder-webpack4') ||
+      dependenciesToRemove.includes('@storybook/manager-webpack4')
+    ) {
+      disclaimer = dedent`\n\n
+      ${chalk.underline(chalk.bold(chalk.cyan('Webpack4 users')))}
+
+      Unless you're using Storybook's Vite builder, this automigration will install a Webpack5-based framework.
+      
+      Given you were using Storybook's Webpack4 builder (default in 6.x, discontinued in 7.0), this could be a breaking change -- especially if your project has a custom webpack configuration.
+      
+      To learn more about migrating from Webpack4, see: ${chalk.yellow(
+        'https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#webpack4-support-discontinued'
+      )}`;
+    }
+
+    if (allDependencies.next && semver.gte(semver.coerce(allDependencies.next).version, '12.0.0')) {
+      if (dependenciesToRemove.includes('storybook-addon-next-router')) {
+        migrationSteps += `- Migrate the usage of the ${chalk.cyan(
+          'storybook-addon-next-router'
+        )} addon to use the APIs from the ${chalk.magenta(
+          '@storybook/nextjs'
+        )} framework package instead. Follow the instructions below.`;
+      }
+
+      if (frameworkPackage === '@storybook/react-vite') {
+        disclaimer = dedent`\n\n
+          ${chalk.bold('Important')}: We've detected you are using Storybook in a Next.js project.
+  
+          This migration is set to update your project to use the ${chalk.magenta(
+            '@storybook/react-vite'
+          )} framework, but Storybook provides a framework package specifically for Next.js projects: ${chalk.magenta(
+          '@storybook/nextjs'
+        )}.
+  
+          This package provides a better experience for Next.js users, however it is only compatible with the Webpack 5 builder, so we can't automigrate for you, as you are using the Vite builder.
```

Comment:
Subtle, but please try to be consistent about spacing: `Webpack 4`, `Webpack4`, `Webpack 5`